tests/validation/cattlevalidationtest/core/test_machine.py

```diff
@@ -0,0 +1,264 @@
+                                          }
+                   }
+    # Create a Digital Ocean Machine with invalid access token
+    machine = client.create_machine(**create_args)
+
+    machine = wait_until_expected_state(client, machine, "error",
+                                        timeout=DEFAULT_TIMEOUT)
+
+    assert machine.state == 'error'
+    assert error_msg_invalid_region in machine.transitioningMessage
+
+    hosts = machine.hosts()
+    assert len(hosts) == 0
+
+    machine = client.wait_success(machine.remove())
+    assert machine.state == 'removed'
+
+
+def digital_ocean_machine_life_cycle(client, configs, expected_values):
+
+    # Create a Digital Ocean Machine
+    machine = client.create_machine(**configs)
+
+    machine = client.wait_success(machine, timeout=DEFAULT_TIMEOUT)
+    assert machine.state == 'active'
+
+    # Wait until host shows up with some physicalHostId
+    machine = wait_for_host(client, machine)
+    host = machine.hosts()[0]
+    assert host.state == 'active'
+    assert machine.accountId == host.accountId
+
+    # Check that the droplet that is being created in Digital Ocean has the
+    # correct configurations
+
+    droplet = check_host_in_digital_ocean(host.ipAddresses()[0].address)
+
+    assert droplet is not None
+    assert droplet["name"] == machine.name
+    assert droplet["image"]["slug"] == expected_values["image"]
+    assert droplet["size_slug"] == expected_values["size"]
+    assert droplet["region"]["slug"] == expected_values["region"]
+
+    # Remove the machine and make sure that the host
+    # and the machine get removed
+
+    machine = client.wait_success(machine.remove())
+    assert machine.state == 'removed'
+
+    host = client.reload(machine.hosts()[0])
+    assert host.state == 'removed'
+
+    wait_for_host_destroy_in_digital_ocean(host.ipAddresses()[0].address)
+
+
+def wait_for_host(client, machine, timeout=DEFAULT_TIMEOUT):
+    start = time.time()
+    time_elapsed = 0
+    machine = client.reload(machine)
+    while len(machine.hosts()) < 1:
+        time.sleep(1)
+        machine = client.reload(machine)
+        time_elapsed = time.time() - start
+        if time_elapsed > timeout:
+            print str(time_elapsed) + " seconds"
+            raise Exception('Timeout waiting for host to be created.')
+
+    host = machine.hosts()[0]
+    host = wait_until_expected_state(client, host, "active")
+    return machine
+
+
+def check_host_in_digital_ocean(ipaddress):
+    url = 'https://api.digitalocean.com/v2/droplets'
+    headers = {'Authorization': "Bearer " + access_key}
+    r = requests.get(url, headers=headers)
+    response = r.json()
+    r.close()
+    droplet_list = response["droplets"]
+    matched_droplet = None
+
+    for droplet in droplet_list:
+        if droplet["networks"]["v4"][0]["ip_address"] == ipaddress:
+            matched_droplet = droplet
+            break
+
+    return matched_droplet
+
+
+def delete_host_in_digital_ocean(name):
```

Comment:
You might want to wrap the contents of this function in a try/except so that if you need to call it more than once in a test, a failure when deleting machine A doesnt prevent you from attempting to delete machine B